File: Code/Chapter 8/C#/Exercise 8.5/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace Exercise_8_5
{
   partial class Form1 : Form
   {
      private static string dataFilePath = @"..\..\..\..\..\Data.Xml";

      public Form1()
      {
         InitializeComponent();
      }

      private void Form1_Load(object sender, EventArgs e)
      {
         DataSet customerProducts = CreateDataSet.DataSetFiller.FillStrongDataSet(dataFilePath);
         DataTable productsTable = customerProducts.Tables[1];

         DataColumn totalPrice = new DataColumn("Total Price");
         totalPrice.Expression = "Price + Price * TaxPercent";
         productsTable.Columns.Add(totalPrice);
         dataGridView1.DataSource = productsTable;
      }
   }
}

File: Code/Chapter 8/VB.NET/Exercise 8.6/Form1.vb
Imports System.Data
Imports CreateDataSet

Public Class Form1
   Private dataFilePath As String = "..\..\..\..\Data.Xml"
   Private productsTable As DataTable

   Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
      Dim customerProducts As DataSet = CreateDataSet.DataSetFiller.FillStrongDataSet(dataFilePath)
      productsTable = customerProducts.Tables(1)

      Dim totalPrice As DataColumn = New DataColumn("Total Price")
      totalPrice.Expression = "Price + Price * TaxPercent"
      productsTable.Columns.Add(totalPrice)
      dataGridView1.DataSource = productsTable
   End Sub

   Private Sub btnSumPrices_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSumPrices.Click
      Dim price As String = productsTable.Compute("Sum(Price)", "Price < 500").ToString()
      lblSumPrice.Text = "The total price is : " & price
   End Sub
End Class
